docs/user/metal/install_upi.md

```diff
@@ -0,0 +1,326 @@
+# Install: BareMetal User Provided Infrastructure
+
+The steps for performing a UPI-based install are outlined here. Several [Terraform][upi-metal-example] templates are provided as an example to help model your own.
+
+## Table of contents
+
+1. Minimum compute requirements
+
+2. Network topology requirements
+
+3. DNS requirements
+
+4. Getting Ignition configs for machines
+
+5. Getting OS related assets for machines
+
+6. Booting machines with RHCOS and Ignition configs
+
+7. Watching your installation (bootstrap_complete, cluster available)
+
+8. Example Bare-Metal UPI deployment
+
+## Compute
+
+The smallest OpenShift 4.x clusters require the following host:
+
+* 1 bootstrap machine.
+
+* 3 control plane machines.
+
+* at least 1 worker machine.
+
+NOTE: The cluster requires the bootstrap machine to deploy the OpenShift cluster on to the 3 control plane machines, and you can remove the bootstrap machine.
+
+The bootstrap and control plane machines must use Red Hat Enterprise Linux CoreOS (RHCOS) as the operating system.
+
+### Minimum resource requirements
+
+Processing
+Memory
+Storage
+Networking
+
+[todo-link-to-minimum-resource-requirements]
+
+## Network Topology Requirements
+
+OpenShift 4.x requires all nodes to have internet access to pull images for platform containers and provide telemetry data to Red Hat.
+
+### Load balancers
+
+Before you install OpenShift, you must provision two load balancers.
+
+* A load balancer for the control plane machines that targets port 6443 (Kubernetes APIServer) and 22623([Machine Config server][machine-config-server]). Port 6443 must be accessible to both clients external to the cluster and nodes within the cluster, and port 22623 must be accessible to nodes within the cluster.
+
+* A load balancer for the machines that run the [ingress router][openshift-router] pods that balances ports 443 and 80. Both the ports must be accessible to both clients external to the cluster and nodes within the cluster.
+
+    NOTE: A working configuration for the ingress router is required for an OpenShift 4.x cluster.
+
+    NOTE: The default configuration for Cluster Ingress Operator  deploys the ingress router to `worker` nodes in the cluster. The administrator needs to configure the [ingress][openshift-router] after the control plane has been bootstrapped.
+
+### Connectivity between machines
+
+You must configure the network connectivity between machines to allow cluster components to communicate.
+
+* Etcd
+
+    As the etcd members are located on the control plane machines. Each control plane machine requires connectivity to [etcd server][etcd-ports], [etcd peer][etcd-ports] and [etcd-metrics][etcd-ports] on every other control plane machine.
+
+    All the worker machines should have connectivity to [etcd server][etcd-ports] and [etcd-metrics][etcd-ports] ports on the control plane machines.
```

Comment:
@abhinavdahiya, do you remember why compute machines need access to etcd?  I expect they need etcd-metrics in case they end up running Prometheus (is this #1334), but I don't understand why compute nodes would need direct access to the etcd server itself.  CC @katherinedube

Reply:
We don't need this for IPI AWS, where [we only allow master->master etcd][1].

[1]: https://github.com/openshift/installer/blob/d1cb735dfe24c664463278ed995e10e342024b0d/data/data/aws/vpc/sg-master.tf#L186

Reply:
From @abhinavdahiya, at some point folks were scraping etcd metrics from the server port.  We can remove this once we here from etcd/monitoring that they no longer need to scrape metrics from the server port.

Reply:
Evidence of etcd-metrics getting scraped from their own port:

```console
$ curl -s https://storage.googleapis.com/origin-ci-test/logs/release-openshift-origin-installer-e2e-aws-upgrade/706/artifacts/e2e-aws-upgrade/pods.json | grep -o 'Delete endpoint [0-9:.]* for service ["\\]*openshift-etcd/etcd:etcd-metrics' | sort | uniq
Delete endpoint 10.0.149.41:9979 for service \"openshift-etcd/etcd:etcd-metrics
```

Is that enough for us to pull the plug on recommending this server-port access?

Reply:
Dropping in #1671